src/Network/Bugsnag/BugsnagRequestHeaders.hs

```diff
@@ -0,0 +1,52 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Network.Bugsnag.BugsnagRequestHeaders
+  ( BugsnagRequestHeaders
+  , bugsnagRequestHeaders
+  , bugsnagRequestHeadersToHeaders
+  , mapBugsnagRequestHeaders
+  , redactBugsnagRequestHeaders
+  )
+ where
+
+import Data.Aeson
+import Data.ByteString (ByteString)
+import Data.CaseInsensitive (CI)
+import qualified Data.CaseInsensitive as CI
+import Data.Text (Text)
+import qualified Data.Text.Encoding as TE
+import Network.HTTP.Types
+
+-- | Wrapper around Wai's 'RequestHeaders', used to give a custom ToJSON instance.
+newtype BugsnagRequestHeaders = BugsnagRequestHeaders {unBugsnagRequestHeaders :: RequestHeaders}
+    deriving (Show, Eq, Ord)
```

Comment:
:warning: 

```suggestion
    deriving stock (Show, Eq, Ord)
```

(Might need to turn on the extension, haven't worked in this codebase since most recent GHC made this a warning)